src/models/BibDetails.ts

```diff
@@ -3,18 +3,39 @@ import type {
   LinkedBibDetail,
   BibDetail,
   FieldMapping,
+  AnnotatedMarcField,
+  Url,
+  SubjectHeadingDetail,
+  AnnotatedMarc,
+  AnyBibDetail,
 } from "../types/bibDetailsTypes"
 
-interface BibDetails extends Bib {
-  groupedNotes: object
-}
 export default class BibDetailsModel {
-  bib: BibDetails
-  constructor(bib: Bib) {
+  bib: Bib
+  annotatedMarcFields: AnnotatedMarcField[]
+  constructor(bib: Bib, annotatedMarc?: AnnotatedMarc) {
     this.bib = this.matchParallelToPrimaryValues(bib)
+    this.annotatedMarcFields = annotatedMarc?.fields
   }
 
-  get holdingsDetails() {
+  get annotatedMarcDetails(): AnyBibDetail[] {
```

Comment:
Could annotatedMarcDetails be changed from a getter function to a class property? I was thinking the code below could be moved into a function that parses the annotatedMarc attribute once on instantiation rather than each time the getter is called (see getItemsFromResult in SearchResultsBib)

Reply:
Furthermore, I don't think we would need annotatedMarcFields to be a class property since it could just be passed as an argument in the preprocessor function.

Reply:
updated!